Civi/Api4/Service/Schema/Joiner.php

```diff
@@ -83,11 +83,18 @@ protected function getPath($baseTable, $joinPath) {
           throw new \Exception(sprintf('Cannot join %s to %s', $baseTable, $targetAlias));
         }
         else {
-          $fullPath = array_merge($fullPath, $links);
+          $fullPath[] = $links;
           $lastLink = end($links);
           $baseTable = $lastLink->getTargetTable();
         }
       }
+      
+      if (PHP_VERSION_ID < 50600) {
+        $fullPath = call_user_func_array('array_merge', $fullPath);
+      }
+      else {
+        $fullPath = array_merge(...$fullPath);
+      }
```

Comment:
This really needs a comment to explain what's going on.

Reply:
Everything is described in the official [documentation](http://php.net/manual/en/functions.arguments.php#functions.variable-arg-list) and  [sample code](http://php.net/manual/ru/functions.arguments.php#121579) (also from official documentation)

Reply:
Since 5.6 is now the minimum recommended php version for CiviCRM, I think this conditional can be removed.